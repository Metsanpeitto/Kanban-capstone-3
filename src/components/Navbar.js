import React from 'react';
import { Link } from 'react-router-dom';

const Navbar = () => (
  <nav className="navBar">
    <h1 className="title">Space Travelers` Hub</h1>
    <ul className="menuNav">
      <li key={1}>
        <Link to="/" active="true" exact="true">
          Rockets
        </Link>
      </li>
      <li key={2}>
        <Link to="/dragons" active="true" exact="true">
          Dragons
        </Link>
      </li>
      <li key={3}>
        <Link to="/missions" exact="true">
          Missions
        </Link>
      </li>
      <li key={4}>
        <Link to="/myprofile" exact="true">
          My Profile
        </Link>
      </li>
    </ul>
  </nav>
);
export default Navbar;
