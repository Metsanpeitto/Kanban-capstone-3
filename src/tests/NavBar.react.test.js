import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from '../components/Navbar';

test('renders the navigation bar', () => {
  render(
    <Router basename={process.env.PUBLIC_URL}>
      <Navbar />
    </Router>,
  );
  const linkElement = screen.getByText(/Rockets/i);
  expect(linkElement).toBeInTheDocument();
});
/*
it('renders correctly', () => {
  const tree = renderer
    .create(
      <Router basename={process.env.PUBLIC_URL}>
        <Link to="/" active="true" exact="true">
          Rockets
        </Link>
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
*/
