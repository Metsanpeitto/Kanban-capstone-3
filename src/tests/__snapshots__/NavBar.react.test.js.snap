// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  active="true"
  exact="true"
  href="/"
  onClick={[Function]}
>
  Rockets
</a>
`;
